kubetest2-gke/deployer/deployer.go

```diff
@@ -223,56 +240,84 @@ func (d *deployer) Up() error {
 	if err := d.init(); err != nil {
 		return err
 	}
-	if err := d.prepareGcpIfNeeded(); err != nil {
+
+	// Only run prepare once for the first GCP project.
+	if err := d.prepareGcpIfNeeded(d.projects[0]); err != nil {
 		return err
 	}
-
-	// Create network if it doesn't exist.
-	if runWithNoOutput(exec.Command("gcloud", "compute", "networks", "describe", d.network,
-		"--project="+d.project,
-		"--format=value(name)")) != nil {
-		// Assume error implies non-existent.
-		log.Printf("Couldn't describe network '%s', assuming it doesn't exist and creating it", d.network)
-		if err := runWithOutput(exec.Command("gcloud", "compute", "networks", "create", d.network,
-			"--project="+d.project,
-			"--subnet-mode=auto")); err != nil {
-			return err
-		}
+	if err := d.createNetwork(); err != nil {
+		return err
+	}
+	if err := d.setupNetwork(); err != nil {
+		return err
 	}
 
+	log.Printf("Environment: %v", os.Environ())
+	eg, _ := errgroup.WithContext(context.Background())
 	loc, err := d.location()
 	if err != nil {
 		return err
 	}
-	args := make([]string, len(d.createCommand()))
-	copy(args, d.createCommand())
-	args = append(args,
-		"--project="+d.project,
-		loc,
-		"--machine-type="+d.machineType,
-		"--image-type="+image,
-		"--num-nodes="+strconv.Itoa(d.nodes),
-		"--network="+d.network,
-	)
-	fmt.Printf("Environment: %v", os.Environ())
-
-	args = append(args, d.cluster)
-	fmt.Printf("Gcloud command: gcloud %+v", args)
-	if err := runWithOutput(exec.Command("gcloud", args...)); err != nil {
-		return fmt.Errorf("error creating cluster: %v", err)
+	for i := range d.projects {
+		project := d.projects[i]
+		subNetworkArgs := d.subNetworkArgs(i)
+		for j := range d.clustersLayout[project] {
+			cluster := d.clustersLayout[project][j]
+			eg.Go(func() error {
+				// Create the cluster
+				args := make([]string, len(d.createCommand()))
+				copy(args, d.createCommand())
+				args = append(args,
+					"--project="+project,
+					loc,
+					"--machine-type="+d.machineType,
+					"--image-type="+image,
+					"--num-nodes="+strconv.Itoa(d.nodes),
+					"--network="+d.transformNetworkName(),
+				)
+				args = append(args, subNetworkArgs...)
+				args = append(args, cluster)
+				log.Printf("Gcloud command: gcloud %+v\n", args)
+				if err := runWithOutput(exec.Command("gcloud", args...)); err != nil {
```

Comment:
we might need to introduce functionality that uses https://golang.org/pkg/os/exec/#CommandContext 
in our internal package and pass the contextwithcancel generated above.

Reply:
Done. Made the change together with the above one.